### PR TITLE
Base structure for the C7 Jenkins setup.

### PR DESCRIPTION
**Pull request by dubb-b**
Base setup of Jenkins using the install guides from Jenkins, Nginx, and EPEL. 

This setup will do the following:
base packages for the OS
setup dependancies for Jenkins
setup firewalld 
setup nginix
setup and install jenkins service -- (not configured)

/TODO's are located in comments

### CODE REVIEW

**Comment by rallytime** [comment]
Nice work! This is great to see!

I am happy to meet and talk about any ideas/needs/questions you have about the "jobs" set up, if desired.

**Comment by cachedout** [comment]
Very good!

File: salt/jenkins-server/deps/init.sls [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Any dependencies for Jenkins can go here
+# currently Java 8 is the only one.
+
+{% if grains['os_family'] == 'RedHat'%}
+app.packages:
+  pkg.installed:
+    - pkgs:
+      - java-1.8.0-openjdk
```
Comment (Ch3LL) [inline salt/jenkins-server/deps/init.sls]: Maybe add an option to grab this from pillar?  `{{ salt['pillar.get']('jenkins:deps:java', 'java-1.8.0-openjdk') }}` So its easy to change the version of java without editing the sls files in this repo each time.

File: salt/jenkins-server/jenkins/init.sls [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+validate_jenkins_repo_gone:
+  file.absent:
+    - name: /etc/yum.repos.d/jenkins.repo
+
+validate_master_cache_gone:
+  file.absent:
+    - name: /var/cache/salt/minion/extrn_files/base/pkg.jenkins.io/redhat-stable/jenkins.repo
+
+
```
Comment (Ch3LL) [inline salt/jenkins-server/jenkins/init.sls]: extra line?

Comment (dubb-b) [inline salt/jenkins-server/jenkins/init.sls]: file.absent:
+    - name: /var/cache/salt/minion/extrn_files/base/pkg.jenkins.io/redhat-stable/jenkins.repo
+
+


That was a hack to force this to work... I need to figure out a real solution to the issue.

I can add the java to pillar data, I also found a little bug in nginx already. So Ill have to get that in as well.

**Pull request merged**